build.gradle

```diff
@@ -16,6 +16,10 @@ dependencies {
     implementation 'ch.qos.logback:logback-core:1.2.11'
     implementation 'org.slf4j:slf4j-api:1.7.36'
     implementation 'ch.qos.logback:logback-classic:1.2.11'
+  //  implementation 'org.mockito:mockito-inline:4.6.1'
+    implementation 'org.mockito:mockito-junit-jupiter:4.6.1'
+    implementation 'org.mockito:mockito-all:1.10.19'
+    implementation 'junit:junit:4.13.1'
     testImplementation 'org.junit.jupiter:junit-jupiter-api:5.8.1'
     testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.8.1'
 }
```

Comment:
Just a suggestion to make testing a bit easier.

Try to replace the content of `build.gradle` (this file) with the following:

```
plugins {
    id 'application'
    id 'java'
}

apply plugin : "java"

group 'org.example'
version '1.0-SNAPSHOT'

repositories {
    mavenCentral()
}

dependencies {

    implementation 'org.glassfish.jersey.inject:jersey-hk2:3.0.4'
    implementation 'org.glassfish.jersey.containers:jersey-container-grizzly2-http:3.0.4'
    implementation 'ch.qos.logback:logback-core:1.2.11'
    implementation 'org.slf4j:slf4j-api:1.7.36'
    implementation 'ch.qos.logback:logback-classic:1.2.11'
  //  implementation 'org.mockito:mockito-inline:4.6.1'
    implementation 'org.mockito:mockito-junit-jupiter:4.6.1'
    implementation 'org.mockito:mockito-all:1.10.19'
    implementation 'junit:junit:4.13.1'
    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.8.1'
    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.8.1'
}

ext {
    javaMainClass = "MainApp"
}

application {
    mainClassName = javaMainClass
}

test {
    useJUnitPlatform()
}
```

Then to compile the project and run unit test from the command line run the command:

```
./gradlew build
```

And to start the server and run test it with the `curl` command:

```
./gradlew run
```
The service should now be running on port 8080 and the following commands should work (from command line):

```
curl http://localhost:8080/api/healthz
Hello World Jersey and HK2
```
and
```
curl http://localhost:8080/api/sol?date=2020-01-01
2632
```

If you have other means to achieve the same then you can ignore this comment...

By the way a `curl` request with wrong `date` parameter gives back a rather harsh error message. Example (added the `-v` option to the `curl` command for more information about the response):
```
curl -v http://localhost:8080/api/sol?date=2020
*   Trying 127.0.0.1:8080...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8080 (#0)
> GET /api/sol?date=2020 HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.68.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 500 Request failed.
< Content-Type: text/html;charset=ISO-8859-1
< Connection: close
< Content-Length: 1031
< 
<html><head><title>Grizzly 3.0.0</title><style><!--div.header {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#003300;font-size:22px;-moz-border-radius-topleft: 10px;border-top-left-radius: 10px;-moz-border-radius-topright: 10px;border-top-right-radius: 10px;padding-left: 5px}div.body {font-family:Tahoma,Arial,sans-serif;color:black;background-color:#FFFFCC;font-size:16px;padding-top:10px;padding-bottom:10px;padding-left:10px}div.footer {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#666633;font-size:14px;-moz-border-radius-bottomleft: 10px;border-bottom-left-radius: 10px;-moz-border-radius-bottomright: 10px;border-bottom-right-radius: 10px;padding-left: 5px}BODY {font-family:Tahoma,Arial,sans-serif;color:black;background-color:white;}B {font-family:Tahoma,Arial,sans-serif;color:black;}A {color : black;}HR {color : #999966;}--></style> </head><body><div class="header">Request failed.</div><div class="body">Request failed.</div><div class="footer">Grizzly 3.0.0</div></body>* Closing connection 0
```
That is we get a "500 Internal Error" back, whcih is not good. 

The reason for this is that the exception thrown in `MessageServiceImpl::convertCuriositySol()` is not handled in the resource class (the `MyResource::convertCuriositySol()` method). But fixing this is probably for a later task I guess?

Reply:
@kmmatwork As mentioned above, I changed method **convertCuriositySol** return type form int to Response. The updated code is below.

`@Path("/sol")
  @GET
  @Produces(MediaType.TEXT_PLAIN)
  public Response convertCuriositySol(@QueryParam("date") String date) {
    if(date == null){
      // assign current date
      date = msgService.getTodayDate();
    }
    logger.info("Example log from "+ MyResource.class.getSimpleName() +" class, get method : convertCuriositySol()");
    try {
      return Response.
          status(Response.Status.OK)
          .entity(msgService.convertCuriositySol(date))
          .build();
     // return msgService.convertCuriositySol(date);
    } catch (Exception e) {
      return Response
          .status(Status.BAD_REQUEST)
          .entity(e.getMessage())
          .build();
    }
  }`

I didn't find INVALID_ARGUMENT status type in the recommended list. So I am using BAD_REQUEST with the text message from the exception object.